rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImpl.java

```diff
@@ -23,7 +23,18 @@ public ArtifactDownloaderImpl(ClouddriverService clouddriverService) {
     this.clouddriverService = clouddriverService;
   }
 
-  public void downloadArtifact(Artifact artifact, Path targetFile) throws IOException {
+  public InputStream downloadArtifact(Artifact artifact) throws IOException {
+    Response response =
+        retrySupport.retry(() -> clouddriverService.fetchArtifact(artifact), 5, 1000, true);
+    if (response.getBody() != null) {
+      return response.getBody().in();
+    } else {
+      log.error("Failure to fetch artifact: empty response");
+      return null;
+    }
+  }
+
+  public void downloadArtifactToFile(Artifact artifact, Path targetFile) throws IOException {
```

Comment:
To reduce code duplication, I'd suggest having `downloadArtifactToFile` call the new `downloadArtifact` function above instead of directly calling out to `clouddriverService`.